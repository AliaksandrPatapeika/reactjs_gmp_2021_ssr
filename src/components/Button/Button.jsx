/* eslint-disable */
import PropTypes from 'prop-types';
import React from 'react';

const Button = React.forwardRef(({
  type, id, className, onClick, disabled, title, children
}, ref) => (
  <button type={type} disabled={disabled} id={id} ref={ref} className={className} onClick={onClick}>
    {title}
    {children}
  </button>
));

Button.propTypes = {
  type: PropTypes.string,
  id: PropTypes.string,
  className: PropTypes.string,
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
  title: PropTypes.string,
  children: PropTypes.node
};

Button.defaultProps = {
  type: 'button',
  id: null,
  className: '',
  onClick: () => {},
  disabled: false,
  title: '',
  children: ''
};

export default Button;
