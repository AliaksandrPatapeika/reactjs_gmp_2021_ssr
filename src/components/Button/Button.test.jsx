import {noop} from 'lodash';
import React from 'react';
import renderer from 'react-test-renderer';

import Button from './Button';

describe('Button', () => {
  test('renders correctly', () => {
    const button = renderer
      .create(<Button onClick={noop} title="Title" />)
      .toJSON();

    expect(button).toMatchSnapshot();
  });
});
