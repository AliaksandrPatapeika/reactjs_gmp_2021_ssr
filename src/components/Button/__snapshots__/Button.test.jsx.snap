// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  className=""
  disabled={false}
  id={null}
  onClick={[Function]}
  type="button"
>
  Title
  
</button>
`;
